scripts/login-page.nse

```diff
@@ -0,0 +1,155 @@
+
+portrule = function(host, port)
+  -- Working :
+  -- If -p argument is used, then the script will be executed only on the given ports.
+  -- If -p argument is not used, then the script will be executed on all the open ports.
+  stdnse.debug(string.format("Loading port %d", port.number))
+
+  local selected_ports = shortport.port_or_service({port.number}, "http", "tcp", {"open", "open|filtered"})
+  return selected_ports(host, port)
+end
+
+action = function(host, port)
+
+  local path = "/"
+  local extension = stdnse.get_script_args(SCRIPT_NAME .. ".extension") or "all"
+
+  --  NOTE:
+  --  If any new db file is created for non-existing extension
+  --  make sure to update the name of the extension here also
+  --  inorder to execute this script successfully.
+  local existing_extensions = {
+    "aspx",
+    "asp",
+    "brf",
+    "cgi",
+    "cfm",
+    "js",
+    "jsp",
+    "php"
+  }
+
+  -- Checking whether the user provided extension is existing in our db or not.
+  local flag = 0
+  local string_of_existing_extensions = ''
+  for _, v in ipairs(existing_extensions) do
+    string_of_existing_extensions = string_of_existing_extensions .. v .. ', '
+    if extension == v then
+      flag = 1
+    end
+  end
+
+  string_of_existing_extensions = string_of_existing_extensions .. 'all'
+
+  -- If user provided extension is not in our db
+  if flag == 0 then
+    -- Extension type not found in the db, throwing a suggestion to end user.
+    local err = "There is no extension like '" .. extension .. "' in the db."
+    err = err .. " Send a mail to rewanth1997[at]gmail.com"
+    err = err .. " and it will be updated soon to db."
+    err = err .. " Available extensions are "
+    err = err .. string_of_existing_extensions .. "\n"
+    return err
+  end
+
+  local file = "nselib/data/web-login/" .. extension .. ".lst"
+
+  -- These regex are used while scraping the web pages for confirmation.
+  local regex = {
+    "Username",
+    "Password",
+    "username",
+    "password",
+    "USERNAME",
+    "PASSWORD",
+    "Admin Password"
+  }
+
+  local output = {}
+  local hostname = host.targetname or host.ip
+
+  -- Fetching all the uris from the db
+  local uris = nmap.fetchfile(file)
+  stdnse.debug(string.format("Working on %s", uris))
+
+  -- Reading line by line and sending requests to those pages.
+  for uri in io.lines(uris) do
+    stdnse.debug(string.format("Sending GET request to %s", hostname .. path .. uri))
+
+    local response = http.get(host, port, path .. uri)
+
+    for _, v in ipairs(regex) do
+      if response.body ~= nil and string.match(response.body, v) then
+        local url = hostname .. path .. uri
+        -- Removing the non-alpha numeric strings if there exist any like \x0D
+        local trimmed_url = url:gsub('\x0D','')
+        table.insert(output, trimmed_url)
+        break
+      end
+    end
+
+  end
+
+  -- Counter, that checks if output table is empty or not.
+  local counter = 0

```

Comment:
To check if a table contains anything you can use #table to see the length of the current table/object. Also instead of a standard lua table use stdnse.output_table.